### ISSUE TITLE
Upgrade nan for NodeJS

### ISSUE THREAD

**Issue by jrgriffiniii** [pulibrary/figgy#3726]
Currently using NodeJS release 12.16.1, running `yarn install` yields a series of errors arising from the CPP headers:

```
../../nan/nan_implementation_12_inl.h:103:42: error: no viable conversion from 'v8::Isolate *' to 'Local<v8::Context>'
  return scope.Escape(v8::Function::New( isolate
```

```
../../nan/nan_object_wrap.h:24:25: error: no member named 'IsNearDeath' in 'Nan::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
  assert(persistent().IsNearDeath());
```

```
../../nan/nan_object_wrap.h:124:26: error: no member named 'IsNearDeath' in 'Nan::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
  assert(wrap->handle_.IsNearDeath());
```

https://github.com/nodejs/nan/commit/fbaf42252af279c3d867c6b193571f9711c39847 addresses this upstream, but I have seen discussions referencing the need to upgrade `node-sass` and `node-gyp`.

**Comment by jrgriffiniii** [comment]
After several attempts at upgrading `nan`, `node-sass`, and `node-gyp`, I found that rebuilding with a new `yarn.lock` was the only solution which did not continue to yield build errors.

Unfortunately, this is now blocked by a NodeJS upgrade: https://circleci.com/gh/pulibrary/figgy/24585?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by jrgriffiniii** [comment]
Blocked by https://github.com/pulibrary/princeton_ansible/pull/1303

**Comment by jrgriffiniii** [comment]
The only outstanding problem on this seems to be some node-sass error, but once I resolve this I should be able to issue a PR.

**Comment by jrgriffiniii** [comment]
Upgrading `lux` beyond release 2.0.3 would have also blocked this - the JavaScript functionality of certain components alone did not just change, but there are also CSS changes which will require a separate PR.
